src/modules/auth/auth.service.ts

```diff
@@ -82,13 +89,54 @@ export class AuthService {
     const accessToken = await this.generateAuthToken(payload, 'access');
     const refreshToken = await this.generateAuthToken(payload, 'refresh');
 
+    const refreshTokenEntity = this.createRefreshTokenEntity(refreshToken);
+    await this.refreshTokenRepository.insert(refreshTokenEntity);
+
     return {
       message: 'User logged in successfully',
       accessToken,
       refreshToken,
     };
   }
 
+  async issueNewTokens(refreshToken: string) {
+    const decodedRefreshToken =
+      this.jwtService.decode<AuthTokenPayload>(refreshToken);
+
+    const existingRefreshTokenEntity =
+      await this.refreshTokenRepository.findOne({
+        where: { userId: decodedRefreshToken.sub, token: refreshToken },
+        relations: {
+          user: true,
+        },
+      });
+
+    if (!existingRefreshTokenEntity) {
+      throw new UnauthorizedException('Invalid or expired token');
+    }
+
```

Comment:
Check if token is expired. With expires_at or by handling expiration error thrown from jwt.verify. Delete the token from database or mark it as is_deleted depending on the necessary auth logic.

Reply:
> Check if token is expired. With either expires_at or by handling expiration error thrown from jwt.verify. Delete the token from database or mark it as is_deleted depending on the necessary auth logic.

Done